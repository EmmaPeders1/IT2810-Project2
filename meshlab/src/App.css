.App {
  display: flex;
  flex-flow: column wrap;
  text-align: center;
  height: 100%;
  overflow-x: hidden;
  margin-left: auto;
  margin-right: auto;
}

.search-container{
  flex-flow: row wrap;
  display: flex;
  justify-content: center;
  width: 90%;
  margin-left: auto;
  margin-right: auto;
  gap: 1rem;
  padding: 2rem;
}

.filter-container{
  flex-flow: row wrap;
  display: flex;
  justify-content: center;
  width: 90%;
  margin-left: auto;
  margin-right: auto;
  gap: 1rem;
  padding: 2rem;
}

p{
  color: black;
}

p.error-message{
  color:red;
}

.URL-input{
  width: 30rem;
}
.token {
  width: 190px;
}

.theme-container{
  flex-flow: row wrap;
  display: flex;
  justify-content: center;
  gap: 1rem;
  padding: 2rem;
}

#displayContainer{
  width: 90%;
  margin-left: auto;
  margin-right: auto;
  flex-flow: row wrap;
  display: flex;
  justify-content: center;
}

#footerContainer {
  width: 100%;
  height: 7%;
  background-color: blueviolet;
  text-align: right;
  position:relative;
  left:0;
  bottom:0;
  right:0;
}

#headerContainer {
  width: 100%;
}

@media (max-width: 950px) {
  .URL-input, .token{
    max-width: 300px;
  }
}

@media (max-width: 370px) {
  .URL-input, .token{
    max-width: 190px;
  }
}