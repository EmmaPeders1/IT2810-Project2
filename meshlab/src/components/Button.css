.button-component {
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 0.5rem;
  cursor: pointer;
  padding: 0.3rem 0.8rem;
  border-radius: 5px;
  font-size: inherit;
  width: max-content;
  border: 0;
  background: skyblue;
  outline: 1.5px solid rgb(110, 188, 219);
  color: rgb(4, 52, 71);
  white-space: nowrap;
}

.button-component:disabled {
  filter: grayscale(80%);
  cursor: not-allowed;
}

.button-component:focus:not(:disabled),
.button-component:hover:not(:disabled) {
  background: rgb(110, 188, 219);
  outline: 1.5px solid rgb(4, 52, 71);
}

[data-type="danger"] {
  background-color: red;
  color: whitesmoke;
}

[data-type="danger"]:focus:not(:disabled),
[data-type="danger"]:hover:not(:disabled) {
  outline: 1.5px solid rgb(113, 8, 8);
  background-color: rgb(189, 20, 20);
}