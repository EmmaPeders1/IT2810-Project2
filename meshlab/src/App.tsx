import React, { useState } from 'react';
import './App.css';
import Button from './components/Button';
import Input from './components/Input';
import { faSearch, faFilter, faUser, faPen, faExclamationTriangle, faWandMagicSparkles } from '@fortawesome/free-solid-svg-icons';
import { UserFetcher } from './components/UserFetcher';
import { CommitFetcher } from './components/CommitFetcher';
import { IssueFetcher } from './components/IssueFetcher';
import { ProjectContext } from './context/ProjectContext';
import { Filters } from './components/Filters';
import UserFilter from './components/UserFilter';
import Header from './components/Header';
import DatePicker from './components/DatePicker';
import StatusFilter from './components/StatusFilter';
import useLocalStorage from './web-storage/Localstorage';
import styled from 'styled-components';
import { ThemeProvider, createTheme } from '@mui/material/styles';
import CssBaseline from '@mui/material/CssBaseline';
import { PaletteMode } from '@mui/material';
import { ClassNames } from '@emotion/react';
import { Wrapper } from './components/Wrapper';

type dState = "users" | "commits" | "issues" | null;

function App() {

  const [theme, setTheme] = useLocalStorage("theme", "dark");

  //Currently some problem that might be here or in the fetcher components
  //even though submiturl updates and useEffect reruns, the new fetch is not reflected in the ui unless the user 
  //switches what component to display and back again
  const [displayComponent, setDisplayComponent] = useState<dState>(null)
  const display = (displayComponent: dState) => {
    switch (displayComponent) {
      case "users":
        return <UserFetcher />
      case "commits":
        return <CommitFetcher />
      case "issues":
        return <IssueFetcher />
      default:
        return ""
    }
  }

  interface PInfo {
    url: string,
    token: string
  }

  //just defaulting to use our project, only while developing
  const [projectInfo, setProjectInfo] = useState<PInfo>({
    url: 'https://gitlab.stud.idi.ntnu.no/it2810-h22/Team-17/project2',
    token: 'glpat-Fy8Cs4SqsPRrBa6MirZy'
  }
  )

  const [currentURL, setCurrentURL] = useState<string>("");

  const [currentToken, setCurrentToken] = useState<string>("");

  function handleURLChange(e: React.ChangeEvent<HTMLInputElement>) {
    setCurrentURL(e.target.value);
  }

  function handleTokenChange(e: React.ChangeEvent<HTMLInputElement>) {
    setCurrentToken(e.target.value);
  }

  function handleSubmit(event: React.MouseEvent<HTMLButtonElement>) {
    event.preventDefault();
    setProjectInfo({ url: currentURL, token: currentToken })
  }

  return (
    <div className="App" >
      <Wrapper theme={theme}>
      <Header ></Header>
      <p>Everything you could ever want from a repository, right at your fingertips
      </p>
      < div className='theme-container'>
        <Button 
            onClick={() =>
              theme === "light" ? setTheme("purple") : setTheme("light")
            }
            label="Change theme"
            className="change-theme-button"
            icon={faWandMagicSparkles}
          />
      </div>
      <div className="search-container">
        <Input
          className='URL-input'
          onChange={handleURLChange}
          placeholder="Insert your gitlab project URL"
        />
        <Input
          onChange={handleTokenChange}
          placeholder="Insert access token"
        />
        <Button
          onClick={()=> handleSubmit}
          label=" GET "
          className="search-button"
          icon={faSearch}
          onKeyDown={() => console.log("search!")}
        />
      </div>
      <div className="filter-container">
        <Button
          onClick={() => setDisplayComponent("users")}
          label=" USERS "
          className="user-filter-button"
          icon={faUser}
        />
        <Button
          onClick={() => setDisplayComponent("commits")}
          label=" COMMITS "
          className="commit-filter-button"
          icon={faPen}
        />
        <Button
          onClick={() => setDisplayComponent("issues")}
          label=" ISSUES "
          className="issue-filter-button"
          icon={faExclamationTriangle}
        />
      </div>

      <div>
        <ProjectContext.Provider value={projectInfo}>
          {display(displayComponent)}
        </ProjectContext.Provider>
      </div>
      </Wrapper>
    </div >
  );
}



export default App;


