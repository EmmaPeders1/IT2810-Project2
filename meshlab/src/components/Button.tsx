import React from 'react';
import { IconDefinition } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { FC } from 'react';

import './css/Button.css';

interface ButtonProps {
  children?: JSX.Element;
  disabled?: boolean;
  label?: string;
  icon?: IconDefinition;
  type?: string;
  className?: string;
  onClick: (e: React.MouseEvent) => void;
  onKeyDown?: (e: React.KeyboardEvent<HTMLButtonElement>) => void;
}

const Button: FC<ButtonProps> = ({
  children,
  disabled,
  label,
  icon,
  type,
  className, onClick,onKeyDown }) => {
  return (
    <button
      disabled={disabled}
      className={`button-component ${className}`}
      data-type={type}
      onClick={onClick}
      onKeyDown={onKeyDown}
    >
      {icon && <FontAwesomeIcon icon={icon} />}
      {label || children }
    </button>
  );
};

export default Button;