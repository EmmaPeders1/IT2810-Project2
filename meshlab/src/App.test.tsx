import React from 'react';
import renderer from 'react-test-renderer';
import Button from './components/Button';
import App from './App';

test('<Button className="search-button"/> renders correctly', () => {
  const tree = renderer.create(
    <Button onClick={function (e: React.MouseEvent<HTMLButtonElement, MouseEvent>): void {
      throw new Error('Function not implemented.');
    }}
    className="search-button"></Button>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

test('<Button className="user-filter-button"/> renders correctly', () => {
  const tree = renderer.create(
    <Button onClick={function (e: React.MouseEvent<HTMLButtonElement, MouseEvent>): void {
      throw new Error('Function not implemented.');
    }}
    className="user-filter-button"></Button>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

test('<Button className="commit-filter-button"/> renders correctly', () => {
  const tree = renderer.create(
    <Button onClick={function (e: React.MouseEvent<HTMLButtonElement, MouseEvent>): void {
      throw new Error('Function not implemented.');
    }}
    className="commit-filter-button"></Button>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

test('<Button className="issue-filter-button"/> renders correctly', () => {
  const tree = renderer.create(
    <Button onClick={function (e: React.MouseEvent<HTMLButtonElement, MouseEvent>): void {
      throw new Error('Function not implemented.');
    }}
    className="issue-filter-button"></Button>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});
