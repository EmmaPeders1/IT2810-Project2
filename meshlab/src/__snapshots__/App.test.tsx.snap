// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button className="commit-filter-button"/> renders correctly 1`] = `
<button
  className="button-component commit-filter-button"
  onClick={[Function]}
/>
`;

exports[`<Button className="issue-filter-button"/> renders correctly 1`] = `
<button
  className="button-component issue-filter-button"
  onClick={[Function]}
/>
`;

exports[`<Button className="search-button"/> renders correctly 1`] = `
<button
  className="button-component search-button"
  onClick={[Function]}
/>
`;

exports[`<Button className="user-filter-button"/> renders correctly 1`] = `
<button
  className="button-component user-filter-button"
  onClick={[Function]}
/>
`;
